File: csharp/resistor-color-duo/ResistorColorDuo.cs
using System.Linq;
using System.Text;

public static class ResistorColorDuo
{
    private static readonly string[] color_list = {
        "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
    };

    public static int Value(string[] colors)
    {
        var sb = new StringBuilder();

        foreach(var color in colors.Take(2))
        {
            sb.Append(ColorCode(color));
        }

        return int.Parse(sb.ToString());
    }

    public static int ColorCode(string color)
    {
        for (int index = 0; index < color_list.Length; index++)
        {
            if (color == color_list[index])
            {
                return index;
            }
        }

        return -1;
    }

    public static string[] Colors() => color_list;
}

File: typescript/resistor-color-duo/resistor-color-duo.ts
export enum Colors {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white,
}

type Color = keyof typeof Colors

export function decodedValue(colors: Color[]): number{
  const [firstColor, secondColor] = colors;
  const decimal: number = Colors[firstColor] * 10;
  const unit: number = Colors[secondColor];

  return decimal + unit;
}
